geonetwork/gn_api.py

```diff
@@ -184,5 +184,26 @@ def delete_thesaurus_dict(self, name):
         response.raise_for_status()
         return response.json()
 
+    def search(self, query):
+        """
+        Use geonetwork API to search metadata
+        :param query: query generated by frontend app like datahub of geonetwork
+        looks like that :
+         {"query":{"bool":{"must":[{"terms":{"isTemplate":["n"]}},{"multi_match":{"query":"test","type":"bool_prefix","fields":["resourceTitleObject.*^4","resourceAbstractObject.*^3","tag^2","resourceIdentifier"]}}],"must_not":{"terms":{"resourceType":["service","map","map/static","mapDigital"]}}}},"_source":["resourceTitleObject","uuid"],"from":0,"size":20}
+        """
+        headers = {
+            'Accept': 'application/json',
```

Comment:
This line can be removed, since `application/json` is already set by default in the headers, so it does not have to be specified again at each API call:
https://github.com/camptocamp/python-geonetwork/blob/82e3b49a80512e64e826947da3f2e79ab5cb756b/geonetwork/gn_api.py#L33

the base headers and additional headers are merged here:
https://github.com/camptocamp/python-geonetwork/blob/82e3b49a80512e64e826947da3f2e79ab5cb756b/geonetwork/gn_session.py#L48